src/contracts/libraries/SlashingAccountingUtils.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+/**
+ * @title Library of utilities for calculations related to slashing in EigenLayer
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ */
+library SlashingAccountingUtils {
+
+    uint64 internal constant SHARE_CONVERSION_SCALE = 1e18;
+
+    // TODO: evaluate the correct max and work on ensuring this is enforced
+    // an amount of shares over this will cause overflow when multiplying by `SHARE_CONVERSION_SCALE`
+    uint256 internal constant MAX_VALID_SHARES = type(uint96).max;
+
+    uint256 internal constant BIPS_FACTOR = 10000;
+
+    uint64 internal constant BIPS_FACTOR_SQUARED = 1e8;
+
+    // TODO: explain this better. basically seems like we may need to set some max factor beyond which shares are just zeroed out
+    // at present this is (2^160)/1e18 ~ 1.46e30, which would be reached after ~659 consecutive 10% slashings, or ~15 consecutive 99% slashings
+    uint256 internal constant MAX_SCALING_FACTOR = type(uint256).max / (MAX_VALID_SHARES * SHARE_CONVERSION_SCALE);
+
+    function denormalize(uint256 shares, uint64 scalingFactor) internal pure returns (uint256) {
+        return (shares * scalingFactor) / SHARE_CONVERSION_SCALE;
+    }
+
+    function normalize(uint256 nonNormalizedShares, uint64 scalingFactor) internal pure returns (uint256) {
+        return (nonNormalizedShares * SHARE_CONVERSION_SCALE) / scalingFactor;
+    }
+
+    // @notice Overloaded version of `scaleUp` that accepts a signed integer shares amount
+    function denormalize(int256 shares, uint64 scalingFactor) internal pure returns (int256) {
+        return (shares * int256(uint256(scalingFactor))) / int256(uint256(SHARE_CONVERSION_SCALE));
+    }
+
+    // @notice Overloaded version of `scaleDown` that accepts a signed integer shares amount
+    function normalize(int256 nonNormalizedShares, uint64 scalingFactor) internal pure returns (int256) {
+        return (nonNormalizedShares * int256(uint256(SHARE_CONVERSION_SCALE))) / int256(uint256(scalingFactor));
+    }
+
+    // TODO: consider possible loss of precision and its consequences
+    // @dev note that rateToSlash is in parts per BIPS_FACTOR_SQUARED, i.e. in parts per 1e8
+    function findNewScalingFactor(uint64 scalingFactorBefore, uint64 rateToSlash) internal pure returns (uint64) {
+        require(rateToSlash != 0, "cannot slash for 0%");
+        require(rateToSlash <= BIPS_FACTOR_SQUARED, "cannot slash more than 100% at once");
+        uint64 scalingFactorAfter;
```

Comment:
wait, since when are scaling factors only uint64's ? 
this change makes `MAX_SCALING_FACTOR` incorrect!
this also makes overflow of scaling factors possible, and drastically reduces the repeated slashing events to get to the (appropriate) `MAX_SCALING_FACTOR`.
@gpsanant I am assuming you made this change -- we should discuss!